atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -142,6 +143,8 @@ String derivedUserAgent() {
         return userAgent().orElse(callingClass().map(UserAgents::fromClass).orElse(UserAgents.DEFAULT_USER_AGENT));
     }
 
+    abstract ScheduledExecutorService executor();
```

Comment:
Should probably name this more specifically as scheduledExecutor or even lockRefreshingExecutor

Reply:
If feels pretty random to just have this one executor passed in when we have many others that are just created internally. What's the reason for this?

Reply:
I think there was a plan to shift towards a model where we pass executors around, so that internally at least we can control overall thread-pool sizes. Don't really have full context though!